streams/src/test/java/org/apache/kafka/streams/kstream/MaterializedTest.java

```diff
@@ -51,4 +54,21 @@ public void shouldThrowNullPointerIfKeyValueBytesStoreSupplierIsNull() {
     public void shouldThrowNullPointerIfSessionBytesStoreSupplierIsNull() {
         Materialized.as((SessionBytesStoreSupplier) null);
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowIllegalArgumentExceptionIfRetentionIsNegative() {
+        Materialized.as("valid-name").withRetention(Duration.of(-1, ChronoUnit.DAYS));
+    }
+
+    @Test(expected = TopologyException.class)
+    public void shouldThrowTopologyExceptionIfStoreNameExceedsMaxAllowedLength() {
+        final StringBuffer invalidStoreNameBuffer = new StringBuffer();
+        final int maxNameLength = Named.MAX_NAME_LENGTH;
+
+        for (int i = 0; i < maxNameLength + 1; i++) {
+            invalidStoreNameBuffer.append('a');
+        }
+
+        Materialized.as(invalidStoreNameBuffer.toString());
```

Comment:
As above: use `assertThrows` and verify error message

Reply:
Used assertThrows instead of expected annotation. Validated exception error message as well.